schedule/views.py

```diff
@@ -8,25 +7,41 @@
 from forecaster.forecaster import forecast as c2alg2 # company 2 alg 2
 
 from c1algo1 import scheduler as c1alg1# company 1 alg 1
-from c1algo2.forecaster import forecast as c1algo2
+from c1algo2.forecaster import forecast as c1alg2
 
+from schedule.alg_data_generator import get_historic_course_data
+from schedule.alg_data_generator import get_program_enrollment_data
+from schedule.alg_data_generator import get_schedule
+from schedule.alg_data_generator import get_professor_dict_mock
+from schedule.alg_data_generator import get_schedule_alg1_mock
+from schedule.alg_data_generator import get_schedule_alg2_mock
+
+import traceback
+import json
+
+from courses.models import Course
 
 
 class Schedule(APIView):
     # GET / schedule / {year - semester}
     def get(self, request: HttpRequest, year: int, semester: str, requested_company_alg: int) -> HttpResponse:
-        alg2_result = c2alg1(None, None, None) if requested_company_alg == 1 else c2alg2(None, None, None)
-        alg1_result = c1alg1.generate_schedule() if requested_company_alg == 1 else c2alg1(None, None, None)
-
-        # these 2 lines are temporary and just to make unit tests pass while integration is in progress
-        alg2_result = "OK"
-        alg1_result = "OK"
-
-        if "OK" == alg1_result and "OK" == alg2_result:
-            body = "GENERATED SCHEDULE FROM COMPANY %d ALGORITHM, AND BOTH RETURNED OK" % requested_company_alg
-            return HttpResponse(body, status=status.HTTP_200_OK)
-        else: 
-            return HttpResponse(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        # Create params for algorithms packages
+        historical_data = get_historic_course_data()
+        previous_enrollment = get_program_enrollment_data()
+        schedule = get_schedule_alg2_mock()
+        professors = get_professor_dict_mock()
+        schedule_1 = get_schedule_alg1_mock()
+
+        try:
+            schedule = c1alg2(historical_data, previous_enrollment, schedule) if requested_company_alg == 1 \
+                 else c2alg2(historical_data, previous_enrollment, None)
+            schedule = c1alg1.generate_schedule(professors, schedule_1, None) if requested_company_alg == 1 \
+                else c2alg1(None, schedule_1, True)
```

Comment:
Same comment as above. If we use the flag here, make sure you can use the same input for our alg 1 as you do for theirs.